src/Orchard.DisplayManagement/Descriptors/ShapeTemplateStrategy/ShapeTemplateBindingStrategy.cs

```diff
@@ -156,15 +153,20 @@ public void Discover(ShapeTableBuilder builder)
                     if (_logger.IsEnabled(LogLevel.Debug))
                     {
                         _logger.LogDebug("Binding {0} as shape [{1}] for feature {2}",
-                            hit.shapeContext.harvestShapeInfo.TemplateVirtualPath,
+                            hit.shapeContext.harvestShapeInfo.RelativePath,
                             iter.shapeContext.harvestShapeHit.ShapeType,
                             feature.Id);
                     }
+
+                    var viewEngine = _shapeTemplateViewEngines
```

Comment:
To make it more optimized please convert it to a `Dictionary<string, IViewEngine>` and add the extension in the `HarvestShapeInfo` so we don't have to recompute it every time the shape is displayed.